src/main/java/com/uid2/shared/attest/UidCoreClient.java

```diff
@@ -71,179 +59,90 @@ public ICloudStorage getContentStorage() {
+        if (httpResponse.statusCode() == 401) {
             LOGGER.info("Initial response from UID2 Core returned 401, performing attestation");
-            attested = true;
-            try {
-                attestInternal();
-            }
-            catch (UidCoreClientException | IOException e) {
-                notifyResponseStatusWatcher(statusCode);
-                throw e;
-            }
-        } else {
-            notifyResponseStatusWatcher(statusCode);
+            attestationTokenRetriever.attestInternal();
+            attestationToken = attestationTokenRetriever.getAttestationToken();
+            httpResponse = sendHttpRequest(path, attestationToken);
         }
-        return attested;
-    }
 
-    private URLConnection sendGet(String url) throws IOException {
-        final URLConnection conn = openConnection(url, "GET");
-        return conn;
+        return Utils.convertHttpResponseToInputStream(httpResponse);
     }
 
-    public void setAllowContentFromLocalFileSystem(boolean allow) {
-        allowContentFromLocalFileSystem = allow;
-    }
-
-    // open connection with auth & attestation headers attached
-    private URLConnection openConnection(String serviceEndpoint, String httpMethod) throws IOException {
-        final URLConnection urlConnection = (proxy == null ? new URL(serviceEndpoint).openConnection() : new URL(serviceEndpoint).openConnection(proxy));
-
-        if(enforceHttps && !(urlConnection instanceof HttpsURLConnection)) {
+    private HttpResponse sendHttpRequest(String path, String attestationToken) throws IOException, InterruptedException {
+        URI uri = URI.create(path);
+        if (this.enforceHttps && !"https".equalsIgnoreCase(uri.getScheme())) {
             throw new IOException("UidCoreClient requires HTTPS connection");
         }
 
-        if (allowContentFromLocalFileSystem && serviceEndpoint.startsWith("file:/tmp/uid2")) {
-            // returns `file:/tmp/uid2` urlConnection directly
-            return urlConnection;
-        }
-
-        final HttpURLConnection connection = (HttpURLConnection) urlConnection;
-        connection.setRequestMethod(httpMethod);
-        connection.setRequestProperty(Const.Http.AppVersionHeader, appVersionHeader);
+        HttpRequest.Builder httpRequestBuilder = HttpRequest.newBuilder()
+                .uri(uri)
+                .GET()
+                .setHeader(Const.Http.AppVersionHeader, appVersionHeader);
 
         if(this.userToken != null && this.userToken.length() > 0) {
-            connection.setRequestProperty("Authorization", "Bearer " + this.userToken);
+            httpRequestBuilder.setHeader("Authorization", "Bearer " + this.userToken);
         }
-
-        final String atoken = this.attestationToken.get();
-        if(atoken != null && atoken.length() > 0) {
-            connection.setRequestProperty("Attestation-Token", atoken);
+        if(attestationToken != null && attestationToken.length() > 0) {
```

Comment:
If the Bearer token or the attestation token are null or empty, there is no point making the call to Core as we know it will fail

Reply:
Keeping this part to be consistent with the previous behaviour.